pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -120,10 +127,13 @@ func NewCloudProvider(ctx context.Context, options cloudprovider.Options) *Cloud
 				options.StartAsync,
 			),
 		),
-		kubeClient: options.KubeClient,
+		sqsProvider:         sqsProvider,
+		eventBridgeProvider: eventBridgeProvider,
```

Comment:
IIUC, the cloudprovider doesn't need these, they're only consumed by the notification/infrastructure controllers.

Reply:
Make sense. We definitely still need the AWS cloudprovider, though, since we need to use the instanceTypeProvider for the ICE cache. Probably best that we keep this as a singleton rather than generating another instance of it